app/src/main/java/com/example/peterchu/watplanner/home/HomePresenter.java

```diff
@@ -95,24 +99,49 @@ public void onFailure(Call<CourseResponse> call, Throwable t) {
-                @Override
-                public void onResponse(Call<CourseScheduleResponse> call,
-                                       Response<CourseScheduleResponse> response) {
-                    Log.d("HomePresenter", response.toString());
-                    List<CourseSchedule> schedules = response.body().getData();
-                    Log.d("HomePresenter", "Number of courses schedules received: " + schedules.size());
-                }
 
-                @Override
-                public void onFailure(Call<CourseScheduleResponse> call, Throwable t) {
-                    // Log error here since request failed
-                    Log.e("HomePresenter", t.toString());
+            if(dbHandler.getSchedulesCount() == 0) {
+                Call<CourseScheduleResponse> scheduleCall = apiInterface.getSubjectCourseSchedules(
+                        "1175", "ECE", Constants.API_KEY
+                );
+                scheduleCall.enqueue(new Callback<CourseScheduleResponse>() {
+                    @Override
+                    public void onResponse(Call<CourseScheduleResponse> call,
+                                           Response<CourseScheduleResponse> response) {
+                        Log.d("HomePresenter", response.toString());
+                        List<CourseSchedule> schedules = response.body().getData();
+                        Log.d("HomePresenter", "Number of courses schedules received: " + schedules.size());
+                        try {
+                            Log.d("HomePresenter", "Trying to add course schedules");
+
+                            final DBHandlerCallback callback = new DBHandlerCallback() {
+                                @Override
+                                public void onFinishTransaction(DatabaseHandler dbHandler) {
+                                    Log.d("DBHandlerCallback", "Finished adding schedules count: " + dbHandler.getSchedulesCount());
+                                }
+                            };
+
+                            dbHandler.addSchedules(schedules, callback);
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                            Log.e("DatabaseHandler", "Failed to add schedules!!");
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<CourseScheduleResponse> call, Throwable t) {
+                        // Log error here since request failed
+                        Log.e("HomePresenter", t.toString());
+                    }
+                });
+            } else {
+                Log.d("MyActivity", "Num schedules in db: " + dbHandler.getSchedulesCount());
+                List<CourseComponent> schedules = dbHandler.getCourseSchedule("ECE", "103");
+                for (CourseComponent c : schedules) {
+                    Log.d("MyActivity", c.toString());
```

Comment:
Is this else statement necessary? Looks like it's just for debugging purposes and not needed for master